### DISCUSSION TITLE
Use a scheme as a body for a http request

### DISCUSSION THREAD

**Discussion by KieranMarien** [stepci/stepci#158] [Q&A] [answered]
Hi, 

I have a a scheme that looks a bit like this
```
components:
  Orders:
    type: array
    items:
      type: object
      properties:
        Currency:
          type: string
          example: EUR
        Customer:
          type: string
          example: "5679099"
        CustomerExternalId:
``` 

And then I have a step that looks like this
```
      - name: insert order
        http:
          url: http://${{env.host}}/order
          method: POST
          headers:
            Authorization: Bearer ${{captures.access_token}}
            Content-Type: application/json
          json:
            schema:
              $ref: "#/components/Orders"
          check:
            status: 200
``` 

But this is the output
```
Request  HTTP 

POST http://localhost:7071/api/order HTTP/1.1
Authorization: Bearer token
Content-Type: application/json

{"$ref":"#/components/Orders"}
Response
``` 

Is it really sending the $ref and scheme litterally as a json? If so how do I make it not do that? Cause I would really like to reuse a scheme for my body but I can't get it to work. Could be that I missed something inn the docs or examples but I'm not sure. If anyone has any advice on this I'd be gratefull!

Thanks in advance!

**Comment by mishushakov** [comment, marked as answer]
This is expected, the request json/body should be defined explicitly

**Comment by sebastianwis** [comment]
Hey Kieran @KieranMarien
Thank you once again for your contribution here! 
It would be amazing to welcome you to our Discord if you haven't joined it yet: 
https://discord.com/invite/KqJJzJ3BTu
This is where we keep users up to date with updates, features and bug resolutions.
